pulsar-client-cpp/tests/ReaderTest.cc

```diff
@@ -54,6 +57,8 @@ TEST(ReaderTest, testSimpleReader) {
         ASSERT_EQ(expected, content);
     }
 
+    producer.close();
```

Comment:
Nit: these are not technically necessary since `client.close()` also takes care of closing all producers/consumers. It won't hurt though.